Pcap++/header/PcapFilter.h

```diff
@@ -243,62 +243,145 @@ namespace pcpp
 	class IPFilter : public IFilterWithDirection
 	{
 	private:
-		std::string m_Address;
+		IPAddress m_Address;
 		std::string m_IPv4Mask;
 		int m_Len;
 		void convertToIPAddressWithMask(std::string& ipAddrmodified, std::string& mask) const;
 		void convertToIPAddressWithLen(std::string& ipAddrmodified) const;
+
+		bool hasMask() const { return !m_IPv4Mask.empty(); }
 	public:
 		/**
-		 * The basic constructor that creates the filter from an IPv4 address and direction (source or destination)
-		 * @param[in] ipAddress The IPv4 address to build the filter with. If this address is not a valid IPv4 address an error will be
-		 * written to log and parsing this filter will fail
+		 * The basic constructor that creates the filter from an IP address string and direction (source or destination)
+		 * @param[in] ipAddress The IP address to build the filter with.
+		 * @param[in] dir The address direction to filter (source or destination)
+		 * @throws std::invalid_argument The provided address is not a valid IPv4 or IPv6 address.
+		 */
+		IPFilter(const std::string& ipAddress, Direction dir) : IPFilter(IPAddress(ipAddress), dir) {}
+
+		/**
+		 * The basic constructor that creates the filter from an IP address and direction (source or destination)
+		 * @param[in] ipAddress The IP address to build the filter with.
 		 * @param[in] dir The address direction to filter (source or destination)
 		 */
-		IPFilter(const std::string& ipAddress, Direction dir) : IFilterWithDirection(dir), m_Address(ipAddress), m_IPv4Mask(""), m_Len(0) {}
+		IPFilter(const IPAddress& ipAddress, Direction dir) : IFilterWithDirection(dir), m_Address(ipAddress), m_IPv4Mask(""), m_Len(0) {}
 
 		/**
 		 * A constructor that enable to filter only part of the address by using a mask (aka subnet). For example: "filter only IP addresses that matches
 		 * the subnet 10.0.0.x"
 		 * @param[in] ipAddress The IPv4 address to use. Only the part of the address that is not masked will be matched. For example: if the address
-		 * is "1.2.3.4" and the mask is "255.255.255.0" than the part of the address that will be matched is "1.2.3.X". If this address is not a
-		 * valid IPv4 address an error will be written to log and parsing this filter will fail
+		 * is "1.2.3.4" and the mask is "255.255.255.0" than the part of the address that will be matched is "1.2.3.X".
 		 * @param[in] dir The address direction to filter (source or destination)
 		 * @param[in] ipv4Mask The mask to use. Mask should also be in a valid IPv4 format (i.e x.x.x.x), otherwise parsing this filter will fail
+		 * @throws std::invalid_argument The provided address is not a valid IPv4 address.
+		 */
+		IPFilter(const std::string& ipAddress, Direction dir, const std::string& ipv4Mask) : IPFilter(IPv4Address(ipAddress), dir, ipv4Mask) {}
```

Comment:
we may not need this constructor, as IPv4Address has a string implicit constructor.

Reply:
True. Documentation slightly differs tho, as the string constructors can throw `invalid_argument` when constructing the IP from string. Should I move the `@throws std::invaild_argument` documentation to IPv4Address' and IPv6Address' documentations?

Reply:
yes, sounds good.

Reply:
Implemented in aa7ae76 and f2f5fa7

Reply:
And reverted f2f5fa7 because it broke when you have an IP address written as a C-string literal.

Reply:
sure.